src/generators/man-page/index.mjs

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+import { writeFile, readFile } from 'node:fs/promises';
+import { join } from 'node:path';
+
+import {
+  convertOptionToMandoc,
+  convertEnvVarToMandoc,
+} from './utils/converter.mjs';
+
+/**
+ * This generator generates a man page version of the CLI.md file.
+ * See https://man.openbsd.org/mdoc.7 for the formatting.
+ *
+ * @typedef {Array<ApiDocMetadataEntry>} Input
+ *
+ * @type {import('../types.d.ts').GeneratorMetadata<Input, string>}
+ */
+export default {
+  name: 'man-page',
+
+  version: '1.0.0',
+
+  description: 'Generates the Node.js man-page.',
+
+  dependsOn: 'ast',
+
+  async generate(input, options) {
+    // Find the appropriate headers
+    const optionsStart = input.findIndex(({ slug }) => slug === 'options');
+    const environmentStart = input.findIndex(
+      ({ slug }) => slug === 'environment-variables-1'
+    );
+
+    if (optionsStart + environmentStart <= 0) {
+      throw new Error('Could not find headers');
+    }
+
+    // Generate the option mandoc
+    let optionsOutput = '';
+    for (let i = optionsStart + 1; i < environmentStart; i++) {
+      const el = input[i];
+      if (el.heading.depth === 3) {
+        optionsOutput += convertOptionToMandoc(el);
+      }
+    }
+
+    // Generate the environment mandoc
+    let envOutput = '';
+    for (let i = environmentStart + 1; i < input.length; i++) {
```

Comment:
Make a function for each of these loops so that they return these variables, and you assign them to the object.

All good if they are functions defined in the same file.